app/src/androidTest/java/ch/epfl/sweng/swenggolf/ShowOfferStatsTest.java

```diff
@@ -0,0 +1,121 @@
+package ch.epfl.sweng.swenggolf;
+
+import android.content.Intent;
+import android.support.test.espresso.intent.rule.IntentsTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v4.app.FragmentTransaction;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import ch.epfl.sweng.swenggolf.database.Database;
+import ch.epfl.sweng.swenggolf.database.DbError;
+import ch.epfl.sweng.swenggolf.database.FakeDatabase;
+import ch.epfl.sweng.swenggolf.database.FilledFakeDatabase;
+import ch.epfl.sweng.swenggolf.database.ValueListener;
+import ch.epfl.sweng.swenggolf.main.MainMenuActivity;
+import ch.epfl.sweng.swenggolf.offer.Offer;
+import ch.epfl.sweng.swenggolf.profile.User;
+import ch.epfl.sweng.swenggolf.statistics.OfferStats;
+import ch.epfl.sweng.swenggolf.tools.FragmentConverter;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static junit.framework.TestCase.fail;
+import static org.hamcrest.Matchers.not;
+
+
+@RunWith(AndroidJUnit4.class)
+public class ShowOfferStatsTest {
+    @Rule
+    public final IntentsTestRule<MainMenuActivity> mActivityRule =
+            new IntentsTestRule<>(MainMenuActivity.class, false, false);
+
+    private User user = FilledFakeDatabase.getUser(0);
+    private Offer offer;
+
+    /**
+     * Set up a fake database, a fake user and launch activity.
+     */
+    @Before
+    public void setUp() {
+        Database.setDebugDatabase(FakeDatabase.fakeDatabaseCreator());
+        Config.setUser(user);
+        mActivityRule.launchActivity(new Intent());
+    }
+
+    @Test
+    public void statsAreDisplayedOnOwnOffer() {
+        offer = FilledFakeDatabase.getOffer(0);
+        FragmentTransaction transaction = mActivityRule.getActivity()
```

Comment:
Maybe you could extract the transaction to clarify the code ?

Reply:
You are absolutely right, I'll do that!

Reply:
Thanks !